std/algorithm/mutation.d

```diff
@@ -622,6 +619,18 @@ if (isArray!A && isSomeChar!(ElementEncodingType!A) &&
     fill(chrs, 'c');
     assert(chrs == [shared(char)('c')]);
 }
+
+@nogc @safe unittest
+{
+    struct Str(size_t len)
+    {
+        private char[len] _data;
+        void opIndexAssign(char value) @safe @nogc
+        {_data[] = value;}
+    }
+    Str!2 str;
+    str.fill(':');
```

Comment:
I think we should also check the result here (as done on the other tests as well), for example:
```d
static immutable res = "::";
assert(str._data == fes);
``